Backend/Backend/Services/SocialService.cs

```diff
@@ -482,4 +501,91 @@ public async Task<bool> RemoveReviewFromPodcastAsync(Guid podcastId, User user)
     }
 
     #endregion
+
+
+    #region Points
+
+    /// <summary>
+    /// Endpoints to gift Points to a particular episode
+    /// </summary>
+    /// <param name="points"></param>
+    /// <param name="episodeId"></param>
+    /// <param name="user"></param>
+    /// <returns></returns>
+    /// <exception cref="Exception"></exception>
+
+    public async Task<string> GiftPoints(int points, Guid episodeId, User user)
+    {
+        // Check whether the point is greater then zero or Not
+        if(points <= 0)
+        {
+            throw new Exception("No of Points should be Greater then zero");
+
+        }
+        // Limit the points to 200
+        if(points > 200)
+        {
+            throw new Exception("You cant gift more then 200 points");
+        }
+
+        // Check whether the person gifting is different person not the podcaster
+        Episode? ep1 = await _db.Episodes.Include(u => u.Podcast).FirstOrDefaultAsync(u => u.Podcast.PodcasterId == user.Id);
```

Comment:
I think you mightve done this check wrong. 
Would this not just check if the guy is a podcaster in general, not the owner of this particular podcast?
I think this wouldve been the solution, unless im mistaken here

 Episode? ep1 = await _db.Episodes.Include(u => u.Id == episodeId).FirstOrDefaultAsync(u => u.Podcast.PodcasterId == user.Id);

Reply:
Yeah you are right i will resolve this is in my commit